WordPress/Classes/ViewRelated/Blog/TimezoneSelectorViewModel.swift

```diff
@@ -0,0 +1,203 @@
+            continentNames.remove(at: manualOffsetIndex)
+            continentNames.append(self.MANUAL_OFFSET)
+        }
+        var timezoneNamesSortedByContinent: [String: [(String, String)]] = [:]
+        for continent in continentNames {
+            let allTimezonesInContinent = result.filter({ $0.continent == continent })
+
+            if continent == self.MANUAL_OFFSET {
+                // sort the UTC strings
+                let allLabelsAndValues = allTimezonesInContinent.map({ (label: $0.label, value: $0.value) })
+                timezoneNamesSortedByContinent[continent] = allLabelsAndValues.sorted(by: { (left, right) -> Bool in
+                    guard let leftNumString = left.value.components(separatedBy: "UTC").last,
+                        let rightNumString = right.value.components(separatedBy: "UTC").last,
+                        let floatLeftNum = Float(leftNumString),
+                        let floatRightNum = Float(rightNumString) else {
+                            return false
+                    }
+                    return floatLeftNum > floatRightNum
+                })
+            } else {
+                timezoneNamesSortedByContinent[continent] = allTimezonesInContinent.sorted(by: { (timezone1, timezone2) -> Bool in
+                    return timezone1.label < timezone2.label
+                }).map({ ($0.label, $0.value) })
+            }
+        }
+        return (headerNames: continentNames, sortedNamesByContinent: timezoneNamesSortedByContinent)
+    }
+
+    /// returns an IndexPath, which is the user's current selection
+    private func getIndexPathToHighlight(timezoneString: String?, manualOffset: NSNumber?, data: ContinentsAndTimezones, allTimezones: [TimezoneInfo]) -> IndexPath? {
+        if let usersTimeZone = timezoneString, !usersTimeZone.isEmpty {
+            for (continentIndex, continent) in data.0.enumerated() {
+                let allTimezonesInContinent = allTimezones.filter({ $0.continent == continent })
+                guard let keyValPair = allTimezonesInContinent.first(where: { $0.value == usersTimeZone }),
+                    let index = data.1[continent]?.index(where: { (labelValPair) -> Bool in
+                        return labelValPair.0 == keyValPair.label
+                    }) else {
+                        continue
+                }
+                let indexPath = IndexPath(row: index, section: continentIndex)
+                return indexPath
+            }
+        } else if let manualOffset = manualOffset,
+            let section = data.0.index(of: self.MANUAL_OFFSET),
+            let dict = data.1[self.MANUAL_OFFSET] {
+            let hoursUTC = manualOffset.intValue
+            let minutesUTC = abs(Int((manualOffset.doubleValue - Double(hoursUTC)) * 60))
+            let utcString: String
+            if minutesUTC == 0 {
+                utcString = String(format: "UTC%+d", hoursUTC)
```

Comment:
As above (see my comment for the SiteSettingsVC), we should avoid doing this string formatting ourselves, and use `DateComponents` / `DateComponentsFormatter`.